prow/cluster/bazel-cache-configmap.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: bazel-cache-nginx-config
+  namespace: ${NAMESPACE}
+data:
+  default.conf: |
+    server {
+      listen       80;
+      server_name  localhost;
+
+      location /cache/ {
+          # The path to the directory where nginx should store the cache contents.
+          root /var/cache/bazel;
+          # Allow PUT
+          dav_methods PUT;
+          # Allow nginx to create the /ac and /cas subdirectories.
+          create_full_put_path on;
+          # The maximum size of a single file.
+          client_max_body_size 1G;
```

Comment:
is 1G (always) enough? I'm curious if there is a cost to erring on the low end here (e.g. what happens when attempting to write an blobs that exceeds the size threshold, is that a costly thing in terms of time that would repeat on each occurrence?).

Reply:
TBH, I have no idea. Do you think we should increase this number?

Reply:
I think that doubling this would allow us to fit the worst case with some headroom (a fully instrumented Envoy binary).

Reply:
Yeah, I realized it caches even the full binary. I found in my local cache a blob that is 1.4GB in size, which is the proxy with debug symbols.

Reply:
So I just checked with gcc & address sanitizer enabled, the resulting binary will be a whopping `1.9GB`. 
So perhaps capping the cache object size at 4GB is wise, leaving sufficient headroom (I don't think we'll be aware of any cache declines because of objects exceeding the max object size limit once this goes into prod, so chances are we will never go back and tune this?).

Reply:
So I just checked with gcc & address sanitizer enabled, the resulting binary will be a whopping `1.9GB`. 
So perhaps capping the cache object size at 4GB is wise, leaving sufficient headroom (I don't think we'll be aware of any cache declines because of objects exceeding the max object size limit once this goes into prod, so chances are we will never go back and tune this?).

Reply:
perhaps we should just remove this flag?

Reply:
When we would do that, it looks like the setting will default to a low value, 1MB:
https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size

Setting it to 0 would configure the cache to not check object size.. I'm not enthusiastic about doing that, but until we figure out a way to detect & monitor write failures I lean towards agreeing with you on that.

This triggers another thought though -- does this cache evict items to reclaim space when needed? Is unbounded cache growth something we need to worry about?

Reply:
Thanks for checking, this was on my todo list for tomorrow. Regarding the cleanup, I think we are going to do what we do with current cache: a daily cron job that removes old files. Daniel might have other ideas.

Reply:
I haven't tried the combination, but traffic server comes with a http cache that acts like a ring buffer. So at some point it will wrap around and start overwriting earlier cache entries, starting with the oldest one. Might be a good fit for the job, perhaps.

Reply:
IIRC there was something in the bazel docs about limiting cache size. but not sure that applies when using remote cache

Reply:
Updated the size to 4GB